cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -120,6 +120,7 @@ func Test_githubRepository_newGitHubRepository(t *testing.T) {
 			g.Expect(err).NotTo(HaveOccurred())
 			g.Expect(gitHub).To(Equal(tt.want))
 		})
+		resetCaches()
```

Comment:
I would move the `resetCaches()` to be right after `g:=NewWithT(t)`. I think each table test should be independent and shouldn't rely on other tests.